flyteadmin/auth/auth_context.go

```diff
@@ -29,7 +29,7 @@ const (
 )
 
 var (
-	callbackRelativeURL = config.MustParseURL("/callback")
+	callbackRelativeURL = config.MustParseURL("callback")
```

Comment:
I believe this is necessary, because when Flyteadmin generates the callback URL, the logic looks like `URL("foo.tech").ResolveReference(URL("/callback")) -> "foo.tech/callback"`

In fact, the base URL can have any path after it, but the result is the same. `URL("foo.tech/any/path").ResolveReference(URL("/callback")) -> "foo.tech/callback"`

This causes issues when flyte is configured on a custom subpath. To fix this...

By adding `foo.tech/custom-subpath/login` as an AuthorizedURI, and changing the reference to "callback", the logic becomes `URL("foo.tech/custom-subpath/login").ResolveReference(URL("callback")) -> "foo.tech/custom-subpath/callback"`

Reply:
Added the test here https://github.com/flyteorg/flyte/pull/5192/files#diff-aa5ac252b2ec45c95649a3a874e727e436be5c932e3d93b5b2aadcfcc83d57feR12

Although I could also merge this into handlers_test.go, no strong preference.